app/models/user.rb

```diff
@@ -1,10 +1,13 @@
 class User < ApplicationRecord
   has_many :bookings, dependent: :destroy
+  has_many :payments, class_name: "Transaction", foreign_key: "payer_id", dependent: :nullify, inverse_of: :payer
+  has_many :receipts, class_name: "Transaction", foreign_key: "recipient_id", dependent: :nullify,
+    inverse_of: :recipient
 
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable
-  devise :database_authenticatable, :registerable,
-    :recoverable, :rememberable, :validatable, :omniauthable, omniauth_providers: %i[facebook]
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable, :validatable, :omniauthable,
+    omniauth_providers: %i[facebook]
 
   before_save :capitalize_name, unless: :admin?
```

Comment:
I see that you created a one-to-many relationship between User and Transaction. The interesting thing I have learned here is that we can "overwrite" the default Ruby on Rails association understanding: you do not create a Payment model, but you use `class_name` option to state that the User is associated with a Transaction mode. Kudos.
What I would like you to explain me, I didn't quite understand is the option: `inverse_of: recipient`. Why do you need this?

Reply:
I think it means that the opposite of `User.receipts` is `Transaction.recipient`